### ISSUE TITLE
Comparison between MLP and KAN

### ISSUE THREAD

**Issue by huangst21** [CUHK-AIM-Group/U-KAN#3]
Hi, thanks for your valuable work. I am using your libraries and trying to replace KANBlock with MLPblock which is already in the code, the val results for BUSI with default settings are as follows respectively:
`busi_UKAN(KANBlock)
IoU: 0.6252
Dice: 0.7656
HD95: 5.5326`
`busi_UKAN(MLPBlock)
IoU: 0.6247
Dice: 0.7648
HD95: 7.2331`
It looks like the results of my experiment are far from the results of Table 5 in the paper, can you explain the possible reasons? Thanks!

**Comment by XGGNet** [comment]
Thank you for your interest in U-KAN! It is likely that the learning rate --lr was set incorrectly. 

The learning rate reported in the paper is 1e-4, but the default learning rate in the initial version of the code mistakenly set as 1e-2 (this has now been corrected to 1e-4). It should affect a lot on performance. I will endeavor to confirm this promptly and engage in further communication with you.

**Comment by XGGNet** [comment]
The model checkpoints is uploaded. The code's final version was refactored for readability, potentially introducing inconsistencies with the original. Plz feel free to contact me for further issues.

**Comment by huangst21** [comment]
I have modified the learning rate to 1e-4 and have run UKAN and UMLP separately as per the code below:
`python train.py --arch UKAN --dataset busi --input_w 256 --input_h 256 --name busi_UKAN  --data_dir ./inputs
`
`python train.py --arch UMLP --dataset busi --input_w 256 --input_h 256 --name busi_UMLP  --data_dir ./inputs
`
`python val.py --name busi_UKAN `
`python val.py --name busi_UMLP `

The val results are:
`busi_UKAN
IoU: 0.6092
Dice: 0.7534
HD95: 6.1631`
`busi_UMLP
IoU: 0.6421
Dice: 0.7809
HD95: 5.4111`
This looks even stranger. Can you guys generously share your results for U-KAN and U-MLP with the same random seed?

**Comment by XGGNet** [comment]
Sorry for it. As there is a re-organization of the code for readability, we need some time to check this issue. We will give feedback as soon as possible.

**Comment by XGGNet** [comment]
I seem to have found the cause of the problem, that is, a failure of self-defined optimizer strategy triggered by parameter name changes. We're quickly verifying it and will confirm it as soon as possible. Thanks for waiting.

**Comment by XGGNet** [comment]
Thank you for waiting. We've uploaded a new version of the code (the changes are mainly in train.py and archs.py).

We tested several times and observed that the KAN-based model outperforms the MLP-based model. At the same time, occasional performance inversions do occur, but averaging results over multiple times (e.g., 3) shows that KAN still performs significantly better.

**Comment by XGGNet** [comment]
> Thanks to the authors for the quick reply. After reading the code, I think the main modification is to assgin different learning rate to the fc layer (KAN/MLP). The U-KAN architecture code appears to be unchanged, except for a few inconsequential modifications. Can you tell me if there are any other important parts of the modification that I am missing? Thanks for your reply!

Thank! I'm actually not sure what has changed in this system, perhaps not much, or perhaps which variable I removed had an effect. But what I can confirm is that with this code, I would run U_KAN significantly higher than the indicated result of 0.609IOU, while U_KAN performs better than the models with MLP on average. I'll further compare the logic of the code a bit more later to see what's going on, it takes time!
